File: backend/src/main/java/edonymyeon/backend/content/post/application/dto/response/MyPostResponse.java
package edonymyeon.backend.content.post.application.dto.response;

import edonymyeon.backend.content.post.domain.Post;
import java.time.LocalDateTime;

public record MyPostResponse(
        Long id,
        String title,
        String image,
        String content,
        LocalDateTime createdAt,
        PostConsumptionResponse consumption,
        ReactionCountResponse reactionCount
) {

    public static MyPostResponse of(Post post, String baseImageUrl, PostConsumptionResponse postConsumptionResponse) {
        return new MyPostResponse(
                post.getId(),
                post.getTitle(),
                post.hasThumbnail() ? baseImageUrl + post.getThumbnailName() : null,
                post.getContent(),
                post.getCreatedAt(),
                postConsumptionResponse,
                new ReactionCountResponse(post.getViewCount(), post.getCommentCount())
        );
    }
}

File: android/app/src/main/java/com/app/edonymyeon/data/dto/response/MyPostResponse.kt
package com.app.edonymyeon.data.dto.response

import com.app.edonymyeon.data.dto.ConsumptionDataModel
import com.app.edonymyeon.data.dto.ReactionCountDataModel
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MyPostResponse(
    @SerialName("id") val id: Long,
    @SerialName("title") val title: String,
    @SerialName("image") val image: String? = null,
    @SerialName("content") val content: String,
    @SerialName("createdAt") val createdAt: String,
    @SerialName("consumption") val consumption: ConsumptionDataModel,
    @SerialName("reactionCount") val reactionCount: ReactionCountDataModel,
)
